### ISSUE TITLE
Filebeat and Apache Module

### ISSUE THREAD

**Issue by ghost** [elastic/beats#4859]
Hello,

I have set in prod filebeat with apache2 module and when I look the log in kibana, I don't see the vhost name. It's a problem if I have multi vhost on a server, and don't see in kibana for which vhost concern the log...

filebeat.modules:
- module: apache2
  access:
    var.paths: ["/var/log/apache2/access.log"]
  error:
    var.paths: ["/var/log/apache2/error.log"]

- module: system

output:
  elasticsearch:
    hosts: ["ELASTIC_SRV_IP:9200"]

the log example from server:
VHOST_NAME:80 109.128.169.134 - - [09/Aug/2017:16:57:28 +0200] "GET /sites/default/files/Albert_0.jpg HTTP/1.1" 200 110294 "http://HOSTNAME" "Mozilla/5.0 (iPhone; CPU iPhone OS 10_3_1 like Mac OS X) AppleWebKit/603.1.30 (KHTML, like Gecko) Mobile/14E304"

and the result in kibana:
{
  "_index": "filebeat-2017.08.09",
  "_type": "log",
  "_id": "AV3HgXTHs-yNO51h7Bdy",
  "_version": 1,
  "_score": null,
  "_source": {
    "@timestamp": "2017-08-09T14:58:46.000Z",
    "offset": 15658203,
    "apache2": {
      "access": {
        "referrer": "http://website/fr/la-carte",
        "response_code": "200",
        "remote_ip": "XXX.XXX.XXX.XXX",
        "geoip": {
          "continent_name": "Europe",
          "country_iso_code": "BE",
          "location": {
            "lon": 4.35,
            "lat": 50.85
          }
        },
        "method": "GET",
        "user_name": "-",
        "http_version": "1.1",
        "body_sent": {
          "bytes": "1883041"
        },
        "url": "/sites/default/files/0.jpg",
        "user_agent": {
          "patch": "3071",
          "major": "59",
          "minor": "0",
          "os": "Android 6.0",
          "build": "125",
          "os_minor": "0",
          "os_major": "6",
          "name": "Chrome Mobile",
          "os_name": "Android",
          "device": "LG-$2"
        }
      }
    },
    "beat": {
      "hostname": "server_name",
      "name": "server_name",
      "version": "5.4.1"
    },
    "input_type": "log",
    "read_timestamp": "2017-08-09T14:58:46.722Z",
    "source": "/var/log/apache2/access.log",
    "type": "log"
  },
  "fields": {
    "@timestamp": [
      1502290726000
    ]
  },
  "sort": [
    1502290726000
  ]
}

Where's the f***ing VHOST name??? 

Thx,

**Comment by tsg** [comment]
You seem to have a custom log format, which doesn't quite fit the ones we support. I assume you are using a LogFormat directive like the one here: https://httpd.apache.org/docs/2.4/logs.html#virtualhost ?

Would it be possible to provide a few sample log lines with your configuration? I can look into supporting it in the module.

**Comment by Globule-Doc** [comment]
which file do you need ? I have copy/paste my config in the first post...

and apache.conf log section:

LogLevel warn
ErrorLog ${APACHE_LOG_DIR}/error.log

#CustomLog "|/usr/bin/logger -p local1.info -t apache2" full

LogFormat "%v:%p %{X-Forwarded-For}i %l %u %t \"%r\" %>s %O \"%{Referer}i\" \"%{User-Agent}i\"" comonvhost
CustomLog ${APACHE_LOG_DIR}/access.log comonvhost

#ErrorLog ${APACHE_LOG_DIR}/error.log
#LogFormat "%v:%p %h %l %u %t \"%r\" %>s %O \"%{Referer}i\" \"%{User-Agent}i\"" vhost_combined
#LogFormat "%h %l %u %t \"%r\" %>s %O \"%{Referer}i\" \"%{User-Agent}i\"" combined
#LogFormat "%h %l %u %t \"%r\" %>s %O" common
#LogFormat "%{Referer}i -> %U" referer
#LogFormat "%{User-agent}i" agent

**Comment by tsg** [comment]
I'd wanted a few resulting log lines, for our testing. You did post one, but it would be helpful to have a couple more for tests.

**Comment by Globule-Doc** [comment]
log apache2:

1.125 Mobile Safari/537.36"
www.SITENAME SITENAME:80 84.192.51.19 - - [10/Aug/2017:09:32:17 +0200] "GET /nl HTTP/1.1" 200 7170 "http://www.SITENAME/nl/het-menu/ontbijt" "Mozilla/5.0 (Linux; Android 5.0; SM-G900F Build/LRX21T) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/59.0.3071.125 Mobile Safari/537.36"
www.SITENAME SITENAME:80 84.192.51.19 - - [10/Aug/2017:09:32:18 +0200] "GET /sites/default/files/css/css_GQy-bUDD5Gd39b2_rNuwBU4sC87LHgQarF69az-mIZY.css HTTP/1.1" 200 4340 "http://www.SITENAME/nl" "Mozilla/5.0 (Linux; Android 5.0; SM-G900F Build/LRX21T) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/59.0.3071.125 Mobile Safari/537.36"
www.SITENAME SITENAME:80 84.192.51.19 - - [10/Aug/2017:09:32:18 +0200] "GET /sites/default/files/js/js_bHLIOwGqxpCbBxCaJyjqBX-1I7neI_BGd-a61lwkHr8.js HTTP/1.1" 200 43860 "http://www.SITENAME/nl" "Mozilla/5.0 (Linux; Android 5.0; SM-G900F Build/LRX21T) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/59.0.3071.125 Mobile Safari/537.36"
www.SITENAME SITENAME:80 84.192.51.19 - - [10/Aug/2017:09:32:18 +0200] "GET /sites/all/modules/globule/lunch_countdown/videos/Moules-NL-cover.png HTTP/1.1" 200 1153033 "http://www.SITENAME/nl" "Mozilla/5.0 (Linux; Android 5.0; SM-G900F Build/LRX21T) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/59.0.3071.125 Mobile Safari/537.36"
www.SITENAME SITENAME:80 84.192.51.19 - - [10/Aug/2017:09:32:19 +0200] "GET /sites/all/themes/bootstrap/bootstrap_lunchgarden/img/list-style.png HTTP/1.1" 200 1618 "http://www.SITENAME/sites/default/files/css/css_07aPZ7S_B5aEi4KdeJvk8Q3HWj_VRbo3Z4nZaDB5E_g.css" "Mozilla/5.0 (Linux; Android 5.0; SM-G900F Build/LRX21T) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/59.0.3071.125 Mobile Safari/537.36"
www.SITENAME SITENAME:80 84.192.51.19 - - [10/Aug/2017:09:32:19 +0200] "GET /sites/default/files/lunch-mobile-app_0.jpg HTTP/1.1" 200 49936 "http://www.SITENAME/nl" "Mozilla/5.0 (Linux; Android 5.0; SM-G900F Build/LRX21T) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/59.0.3071.125 Mobile Safari/537.36"
www.SITENAME SITENAME:80 84.192.51.19 - - [10/Aug/2017:09:32:19 +0200] "GET /sites/default/files/buffet_1.jpg HTTP/1.1" 200 231400 "http://www.SITENAME/nl" "Mozilla/5.0 (Linux; Android 5.0; SM-G900F Build/LRX21T) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/59.0.3071.125 Mobile Safari/537.36"
www.SITENAME SITENAME:80 84.192.51.19 - - [10/Aug/2017:09:32:19 +0200] "GET /sites/all/themes/bootstrap/bootstrap_lunchgarden/img/bg-news.png HTTP/1.1" 200 6051 "http://www.SITENAME/sites/default/files/css/css_07aPZ7S_B5aEi4KdeJvk8Q3HWj_VRbo3Z4nZaDB5E_g.css" "Mozilla/5.0 (Linux; Android 5.0; SM-G900F Build/LRX21T) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/59.0.3071.125 Mobile Safari/537.36"
www.SITENAME SITENAME:80 84.192.51.19 - - [10/Aug/2017:09:32:19 +0200] "GET /sites/default/files/Albert_0.jpg HTTP/1.1" 200 110294 "http://www.SITENAME/nl" "Mozilla/5.0 (Linux; Android 5.0; SM-G900F Build/LRX21T) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/59.0.3071.125 Mobile Safari/537.36"
www.SITENAME SITENAME:80 84.192.51.19 - - [10/Aug/2017:09:32:19 +0200] "GET /sites/all/modules/globule/lunch_countdown/videos/moules_07-2017_NL.mp4 HTTP/1.1" 200 10344685 "http://www.SITENAME/nl" "Mozilla/5.0 (Linux; Android 5.0; SM-G900F Build/LRX21T) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/59.0.3071.125 Mobile Safari/537.36"
www.SITENAME SITENAME:80 84.192.51.19 - - [10/Aug/2017:09:32:19 +0200] "GET /sites/all/modules/globule/lunch_countdown/js/video-js/font/vjs.woff HTTP/1.1" 200 2905 "http://www.SITENAME/sites/default/files/css/css_GQy-bUDD5Gd39b2_rNuwBU4sC87LHgQarF69az-mIZY.css" "Mozilla/5.0 (Linux; Android 5.0; SM-G900F Build/LRX21T) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/59.0.3071.125 Mobile Safari/537.36"
www.SITENAME SITENAME:80 84.192.51.19 - - [10/Aug/2017:09:32:20 +0200] "GET /sites/all/themes/bootstrap/bootstrap_lunchgarden/img/bg-euro.png HTTP/1.1" 200 3740 "http://www.SITENAME/sites/default/files/css/css_07aPZ7S_B5aEi4KdeJvk8Q3HWj_VRbo3Z4nZaDB5E_g.css" "Mozilla/5.0 (Linux; Android 5.0; SM-G900F Build/LRX21T) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/59.0.3071.125 Mobile Safa

with this custom log:

LogLevel warn
ErrorLog ${APACHE_LOG_DIR}/error.log

#CustomLog "|/usr/bin/logger -p local1.info -t apache2" full

LogFormat "%{Host}i %v:%p %{X-Forwarded-For}i %l %u %t \"%r\" %>s %O \"%{Referer}i\" \"%{User-Agent}i\"" full
#LogFormat "%{Host}i %{X-Forwarded-For}i %l %u %t \"%r\" %>s %O \"%{Referer}i\" \"%{User-Agent}i\"" comonvhost
CustomLog ${APACHE_LOG_DIR}/access.log full

#ErrorLog ${APACHE_LOG_DIR}/error.log
#LogFormat "%v:%p %h %l %u %t \"%r\" %>s %O \"%{Referer}i\" \"%{User-Agent}i\"" vhost_combined
#LogFormat "%h %l %u %t \"%r\" %>s %O \"%{Referer}i\" \"%{User-Agent}i\"" combined
#LogFormat "%h %l %u %t \"%r\" %>s %O" common
#LogFormat "%{Referer}i -> %U" referer
#LogFormat "%{User-agent}i" agent

Kibana log from this server:


Time 	_source 
	August 10th 2017, 09:33:50.000	beat.name:cluster-web02-1 @timestamp:August 10th 2017, 09:33:50.000 offset:1,727,763 apache2.access.referrer:http://www.SITENAME/fr apache2.access.response_code:200 apache2.access.remote_ip:193.58.1.131 apache2.access.geoip.continent_name:Europe apache2.access.geoip.city_name:Steenokkerzeel apache2.access.geoip.country_iso_code:BE apache2.access.geoip.region_name:Flemish Brabant Province apache2.access.geoip.location:{ "lon": 4.5167, "lat": 50.9167 } apache2.access.method:GET apache2.access.user_name:- apache2.access.http_version:1.1 apache2.access.body_sent.bytes:6.928KB apache2.access.url:/nl apache2.access.user_agent.major:11 apache2.access.user_agent.minor:0 apache2.access.user_agent.os:Windows 7 apache2.access.user_agent.name:IE apache2.access.user_agent.os_name:Windows 7 apache2.access.user_agent.device:Other beat.hostname:cluster-web02-1 beat.version:5.4.1 input_type:log read_timestamp:2017-08-10T07:33:53.670Z source:/var/log/apache2/access.log type:log _id:AV3LEI8cs-yNO51hh8r8 _type:log _index:filebeat-2017.08.10 _score: -
	August 10th 2017, 09:33:45.000	beat.name:cluster-web02-1 @timestamp:August 10th 2017, 09:33:45.000 offset:1,727,554 apache2.access.referrer:- apache2.access.response_code:200 apache2.access.remote_ip:193.58.1.131 apache2.access.geoip.continent_name:Europe apache2.access.geoip.city_name:Steenokkerzeel apache2.access.geoip.country_iso_code:BE apache2.access.geoip.region_name:Flemish Brabant Province apache2.access.geoip.location:{ "lon": 4.5167, "lat": 50.9167 } apache2.access.method:GET apache2.access.user_name:- apache2.access.http_version:1.1 apache2.access.body_sent.bytes:1.514KB apache2.access.url:/favicon.ico apache2.access.user_agent.major:11 apache2.access.user_agent.minor:0 apache2.access.user_agent.os:Windows 7 apache2.access.user_agent.name:IE apache2.access.user_agent.os_name:Windows 7 apache2.access.user_agent.device:Other beat.hostname:cluster-web02-1 beat.version:5.4.1 input_type:log read_timestamp:2017-08-10T07:33:46.669Z source:/var/log/apache2/access.log type:log _id:AV3LEHuUs-yNO51hh8QP _type:log _index:filebeat-2017.08.10 _score: -
	August 10th 2017, 09:33:41.000	beat.name:cluster-web02-1 @timestamp:August 10th 2017, 09:33:41.000 offset:1,727,364 apache2.access.referrer:http://www.google.be/url?sa=t&rct=j&q=&esrc=s&source=web&cd=1&ved=0ahUKEwjJ0bCyk8zVAhVEJFAKHb2xB7YQgU8IMjAA&url=http%3A%2F%2Fwww.SITENAME%2Ffr&usg=AFQjCNEHd4SbUA643cjccEyqMut9PSFpoA apache2.access.response_code:200 apache2.access.remote_ip:193.58.1.131 apache2.access.geoip.continent_name:Europe apache2.access.geoip.city_name:Steenokkerzeel apache2.access.geoip.country_iso_code:BE apache2.access.geoip.region_name:Flemish Brabant Province apache2.access.geoip.location:{ "lon": 4.5167, "lat": 50.9167 } apache2.access.method:GET apache2.access.user_name:- apache2.access.http_version:1.1 apache2.access.body_sent.bytes:7.193KB apache2.access.url:/fr apache2.access.user_agent.major:11 apache2.access.user_agent.minor:0 apache2.access.user_agent.os:Windows 7 apache2.access.user_agent.name:IE apache2.access.user_agent.os_name:Windows 7 apache2.access.user_agent.device:Other beat.hostname:cluster-web02-1 beat.version:5.4.1 input_type:log read_timestamp:2017-08-10T07:33:43.668Z source:/var/log/apache2/access.log type:log _id:AV3LEGgOs-yNO51hh5jj _type:log _index:filebeat-2017.08.10 _score: -
	August 10th 2017, 09:33:40.000	beat.name:cluster-web02-1 @timestamp:August 10th 2017, 09:33:40.000 offset:1,726,996 apache2.access.referrer:- apache2.access.response_code:200 apache2.access.remote_ip:150.70.173.56 apache2.access.geoip.continent_name:Asia apache2.access.geoip.country_iso_code:JP apache2.access.geoip.location:{ "lon": 139.69, "lat": 35.69 } apache2.access.method:GET apache2.access.user_name:- apache2.access.http_version:1.1 apache2.access.body_sent.bytes:24.887KB apache2.access.url:/fr apache2.access.user_agent.major:8 apache2.access.user_agent.minor:0 apache2.access.user_agent.os:Windows XP apache2.access.user_agent.name:IE apache2.access.user_agent.os_name:Windows XP apache2.access.user_agent.device:Other beat.hostname:cluster-web02-1 beat.version:5.4.1 input_type:log read_timestamp:2017-08-10T07:33:43.668Z source:/var/log/apache2/access.log type:log _id:AV3LEGgOs-yNO51hh5ji _type:log _index:filebeat-2017.08.10 _score: -
	August 10th 2017, 09:33:34.000	beat.name:cluster-web02-1 @timestamp:August 10th 2017, 09:33:34.000 offset:1,726,811 apache2.access.referrer:http://www.SITENAME/nl apache2.access.response_code:200 apache2.access.remote_ip:84.199.187.83 apache2.access.geoip.continent_name:Europe apache2.access.geoip.country_iso_code:BE apache2.access.geoip.location:{ "lon": 4.35, "lat": 50.85 } apache2.access.method:GET apache2.access.user_name:- apache2.access.http_version:1.1 apache2.access.body_sent.bytes:6.102KB apache2.access.url:/nl/het-menu apache2.access.user_agent.major:11 apache2.access.user_agent.minor:0 apache2.access.user_agent.os:Windows 10 apache2.access.user_agent.name:IE apache2.access.user_agent.os_name:Windows 10 apache2.access.user_agent.device:Other beat.hostname:cluster-web02-1 beat.version:5.4.1 input_type:log read_timestamp:2017-08-10T07:33:36.667Z source:/var/log/apache2/access.log type:log _id:AV3LEFSEs-yNO51hh5Gd _type:log _index:filebeat-2017.08.10 _score: -
	August 10th 2017, 09:33:34.000	beat.name:cluster-web02-1 @timestamp:August 10th 2017, 09:33:34.000 offset:1,726,584 apache2.access.referrer:http://www.SITENAME/nl/restaurant apache2.access.response_code:200 apache2.access.remote_ip:81.245.138.171 apache2.access.geoip.continent_name:Europe apache2.access.geoip.city_name:Dendermonde apache2.access.geoip.country_iso_code:BE apache2.access.geoip.region_name:East Flanders Province apache2.access.geoip.location:{ "lon": 4.1167, "lat": 51.

and a log in json format:
{
  "_index": "filebeat-2017.08.10",
  "_type": "log",
  "_id": "AV3LEVJzs-yNO51hiMjM",
  "_version": 1,
  "_score": null,
  "_source": {
    "@timestamp": "2017-08-10T07:34:40.000Z",
    "offset": 1763180,
    "apache2": {
      "access": {
        "referrer": "http://www.SITENAME/fr",
        "response_code": "200",
        "remote_ip": "109.128.44.76",
        "geoip": {
          "continent_name": "Europe",
          "country_iso_code": "BE",
          "location": {
            "lon": 4.35,
            "lat": 50.85
          }
        },
        "method": "GET",
        "user_name": "-",
        "http_version": "1.1",
        "body_sent": {
          "bytes": "418315"
        },
        "url": "/sites/default/files/Lunch%20Garden%20fajitas.jpg",
        "user_agent": {
          "major": "10",
          "minor": "0",
          "os": "iOS 10.3.2",
          "os_minor": "3",
          "os_major": "10",
          "name": "Mobile Safari",
          "os_name": "iOS",
          "device": "iPhone"
        }
      }
    },
    "beat": {
      "hostname": "SERVERNAME",
      "name": "SERVERNAME",
      "version": "5.4.1"
    },
    "input_type": "log",
    "read_timestamp": "2017-08-10T07:34:41.682Z",
    "source": "/var/log/apache2/access.log",
    "type": "log"
  },
  "fields": {
    "@timestamp": [
      1502350480000
    ]
  },
  "highlight": {
    "beat.name": [
      "@kibana-highlighted-field@SERVERNAME@/kibana-highlighted-field@"
    ]
  },
  "sort": [
    1502350480000
  ]
}

**Comment by Globule-Doc** [comment]
I think it's a problem with filebeat and apache module. If I configure logstash on elk, beat input, apache filter and SAME APACHE LOG FORMAT , I see the vhost name in log....


But with filebeat apache module I can't....

**Comment by jeremy-clerc** [comment]
Hey @tsg, after looking a bit into the different files, I found at least 3 places where it could be missing:

`/etc/filebeat/fields.yml` so I modified it to have a vhost field:
```
      fields:
        - name: access
          type: group
          description: >
            Contains fields for the Apache2 HTTPD access logs.
          fields:
            - name: vhost
              type: keyword
              description: >
                Virtual Host.
            - name: remote_ip
              type: keyword
              description: >
                Client IP address.
```
In `module/apache2/access/ingest/default.json`, I updated the GROK patterns to have the vhost:

```
      "patterns":[
        "%{DATA:apache2.access.vhost} %{IPORHOST:apache2.access.remote_ip} - %{DATA:apache2.access.user_name} \\[%{HTTPDATE:apache2.access.time}\\] \"%{WORD:apache2.access.method} %{DATA:apache2.access.url} HTTP/%{NUMBER:apache2.access.http_version}\" %{NUMBER:apache2.access.response_code} (?:%{NUMBER:apache2.access.body_sent.bytes}|-)( \"%{DATA:apache2.access.referrer}\")?( \"%{DATA:apache2.access.agent}\")?",
        "%{DATA:apache2.access.vhost} %{IPORHOST:apache2.access.remote_ip} - %{DATA:apache2.access.user_name} \\[%{HTTPDATE:apache2.access.time}\\] \"-\" %{NUMBER:apache2.access.response_code} -"
        ],
```

In `kibana/{5.x,default}/index-pattern/filebeat.json` I added vhost too, not puting it here as it is a big JSON.

I deleted the index + the index pattern from kibana and re-run: `/usr/share/filebeat/bin/filebeat -c /etc/filebeat/filebeat.yml setup`

But it does not seem to help getting the field.

Any other place where I could find it ? Or could it be something defined at compile time ?

If I know where to find it and can make it work, I will be happy to help by creating a PR.

Thanks

**Comment by ruflin** [comment]
@jeremy-clerc If you are using the beat directly with the full `fields.yml`, what you did above should have updated at least the index pattern. But you also have to delete and reapply the ingest processors to make it work. Best would be to get this directly into filebeat source code as a PR.


For a PR the file you are looking for is: https://github.com/elastic/beats/blob/master/filebeat/module/apache2/access/_meta/fields.yml Assuming you have checked out `beats` you need to run `make update` after modifying this file so it is applied in all necessary places. It would be great if you could open a PR and also add some test log files. Happy to assist there. Feel free to open PR early so we can take it from there step by step.

**Comment by jeremy-clerc** [comment]
I deleted everything, filebeat data, ES templates and indices, and kibana index-pattern, sure not the best way :) But know it is correctly exported.

Thanks for the tips, I created the PR.

**Comment by ruflin** [comment]
Thanks for the PR.

**Comment by coudenysj** [comment]
Hi, so since #5861 is closed, I was wondering if changing the processing of the default beats module is the way to go.

If we just want to add some fields to the logging, do we need to install a separate Logstash service to handle this? Or is fiddling around with the Filebeat fields a good idea?

**Comment by coudenysj** [comment]
Thanks for the repsonse @ruflin.

As a general question: is it a good idea trying to skip the Logstash step? Or should I just add a Logstash node to process this data?

**Comment by ruflin** [comment]
I personally would not introduce an LS just for that. If you have already and LS node in place and route the data anyways through LS it looks a bit different.

The main problem is we don't provide good tooling here yet. In the end modifying the ingest pipeline in elasticsearch or LS is the same thing with the difference that you have 1 piece less in your architecture. So if possible I would go for the ingest pipeline modification.

As usual, the answer is "it depends" ;-)

**Comment by coudenysj** [comment]
Ok, thanks again for the response!

We will change the ingest pipeline, and try to keep it in mind if we upgrade :smile:.

**Comment by webmat** [comment]
Oh jeez I meant to post the following a few days ago...

-------------------------

@coudenysj Note that if you want to leave the generated pipelines unmodified, you can now do that. Ingest Node pipelines can now be chained (see the [pipeline](https://www.elastic.co/guide/en/elasticsearch/reference/current/pipeline-processor.html)).

Idea is:

- Create a custom pipeline that calls out to the default Filebeat module pipeline.
- Add Ingest Node processors to your custom pipeline before or after the call out to the generated Filebeat module. Depending on when in the processing you want to chime in.
- Point all of your Beat outputs to your new custom pipeline.

The result is that now, when upgrading Beats (and therefore the generated pipeline), the only thing you need to do is point your custom pipeline to this new Filebeat pipeline.

**Comment by click** [comment]
> You can change the default module but need to apply it on every update of the Beat which is not nice. We are thinking about ideas on how to make this easier but we don't have a good solution yet.

Debian, Ubuntu, Linux Mint and others (more or less all other Debian-based distributions) requires manual intervention for the other_vhosts_access.log to be parsed correctly - any progress on figuring out how to handle this in filebeat updates yet?

**Comment by wixaw** [comment]
Hello,
It's a classic in the apache logs to get the vhost ( https://httpd.apache.org/docs/2.2/mod/mod_log_config.html )

> Common Log Format with Virtual Host : 
> "%v %h %l %u %t \"%r\" %>s %b"

Would not it be possible to have an option in "modules.d/apache.yml" of type "vhost_enable: true" ?

Thanks 
William

**Comment by jsoriano** [comment]
Fixed by @wixaw in #12778, thanks!
